lib/entities/guess.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class Guess < ValidatedEntity
+  attr_reader :guess_code, :errors
+
+  SECRET_CODE_LENGTH = 4
+  ELEMENT_MIN_VALUE = 1
+  ELEMENT_MAX_VALUE = 6
+
+  HINT = 'hint'
+  PLUS = '+'
+  MINUS = '-'
+
+  def initialize(guess_code)
+    super()
+    @guess_code = guess_code
+  end
+
+  def validate
+    return if hint?
+
+    @errors << fault.secret_code_format unless check_integer
+    @errors << fault.secret_code_length(SECRET_CODE_LENGTH) unless check_length
+    @errors << fault.secret_code_digits_range(ELEMENT_MIN_VALUE, ELEMENT_MAX_VALUE) unless check_digits_range
+  end
+
+  def mark_guess(secret_code)
+    @secret_code = secret_code
+    cloned_code = @secret_code.clone
+    input_digits = convert_input(@guess_code)
```

Comment:
it's very hard to understand.
1) why do you put instance variable as an argument
2) why you don't use memoization in convert_input method
3) why do you send input_digits to params of `exact_match` and `number_match`

Reply:
1 - didn't mention that mistake
2 - because each time it is different and there i no need to memorize it
3 - provided new way of solving this issue